sqlmesh/core/environment.py

```diff
@@ -102,32 +106,50 @@ class Environment(EnvironmentNamingInfo):
         previous_finalized_snapshots: Snapshots that were part of this environment last time it was finalized.
     """
 
-    snapshots: t.List[SnapshotTableInfo]
+    snapshots_: SnapshotTableInfoOrJsonList = Field(alias="snapshots")
     start_at: TimeLike
     end_at: t.Optional[TimeLike] = None
     plan_id: str
     previous_plan_id: t.Optional[str] = None
     expiration_ts: t.Optional[int] = None
     finalized_ts: t.Optional[int] = None
-    promoted_snapshot_ids: t.Optional[t.List[SnapshotId]] = None
-    previous_finalized_snapshots: t.Optional[t.List[SnapshotTableInfo]] = None
-
-    @field_validator("snapshots", "previous_finalized_snapshots", mode="before")
+    promoted_snapshot_ids_: t.Optional[SnapshotIdOrJsonList] = Field(
+        default=None, alias="promoted_snapshot_ids"
+    )
+    previous_finalized_snapshots_: t.Optional[SnapshotTableInfoOrJsonList] = Field(
+        default=None, alias="previous_finalized_snapshots"
+    )
+
+    @field_validator("snapshots_", "previous_finalized_snapshots_", mode="before")
     @classmethod
-    def _convert_snapshots(
-        cls, v: str | t.List[SnapshotTableInfo] | None
-    ) -> t.List[SnapshotTableInfo] | None:
+    def _load_snapshots(
+        cls, v: str | SnapshotTableInfoOrJsonList | None
+    ) -> SnapshotTableInfoOrJsonList | None:
         if isinstance(v, str):
-            return [SnapshotTableInfo.parse_obj(obj) for obj in json.loads(v)]
+            return json.loads(v)
```

Comment:
this is not json, json is a string, this is returning dicts

Reply:
OBE